### ISSUE TITLE
Paper 1.15.2 FastAsyncWorldEdit not working along with procedual damage and snapshops

### ISSUE THREAD

**Issue by LeehamElectronics** [kangarko/CoreArena#323]
I'v been trying to get procedural damage gravity engine working for a while now. I have tried using world edit and AsyncWorldEdit and both of them worked for blowing up blocks with the first and final snapshots but the whole gravity and "flying blocks" does not work, I disabled world guard flags aswell. I have procedural damage enabled in my settings.yml and I thought I would now try FastAsyncWorldEdit but CoreArena returns the following errors when I try to save snapshots:

```
5:26:22 INFO]: CoreArena 2.7.4 encountered an EOFException! Please check your error.log and report this issue with the information in that file.
[15:26:22 INFO]: org.mineacademy.corearena.short. .al: Report / Failed to load INITIAL template for DAFF, see below and report (make sure you have the latest WorldEdit)
[15:26:22 INFO]: at org.mineacademy.corearena.short. .dW.a(SourceFile:107)
[15:26:22 INFO]: at org.mineacademy.corearena.short. .dW.b(SourceFile:62)
[15:26:22 INFO]: at org.mineacademy.corearena.short. .dW.a(SourceFile:119)
[15:26:22 INFO]: at org.mineacademy.corearena.short. .fS.e(SourceFile:33)
[15:26:22 INFO]: at org.mineacademy.corearena.short. .eW$a.a(SourceFile:132)
[15:26:22 INFO]: at org.mineacademy.corearena.short. .aD.a(SourceFile:775)
[15:26:22 INFO]: at org.mineacademy.corearena.short. .aE.onMenuClick(SourceFile:75)
[15:26:22 INFO]: at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor1115.execute(Unknown Source)
[15:26:22 INFO]: at org.bukkit.plugin.EventExecutor.lambda$create$1(EventExecutor.java:69)
[15:26:22 INFO]: at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:80)
[15:26:22 INFO]: at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70)
[15:26:22 INFO]: at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:607)
[15:26:22 INFO]: at java.lang.Thread.run(Unknown Source)
[15:26:22 INFO]: java.io.EOFException: Stream ended prematurely
[15:26:22 INFO]: at net.jpountz.lz4.LZ4BlockInputStream.readFully(LZ4BlockInputStream.java:200)
[15:26:22 INFO]: at net.jpountz.lz4.LZ4BlockInputStream.refill(LZ4BlockInputStream.java:130)
[15:26:22 INFO]: at net.jpountz.lz4.LZ4BlockInputStream.read(LZ4BlockInputStream.java:82)
[15:26:22 INFO]: at com.sk89q.worldedit.extent.clipboard.io.SchematicReader.readCombined(SchematicReader.java:181)
[15:26:22 INFO]: at com.sk89q.worldedit.extent.clipboard.io.SchematicReader.readwrite(SchematicReader.java:202)
[15:26:22 INFO]: at com.sk89q.worldedit.extent.clipboard.io.SchematicReader.read(SchematicReader.java:282)
[15:26:22 INFO]: at com.sk89q.worldedit.extent.clipboard.io.ClipboardReader.read(ClipboardReader.java:59)
[15:26:22 INFO]: at com.sk89q.worldedit.extent.clipboard.io.ClipboardReader.read(ClipboardReader.java:46)
[15:26:22 INFO]: at org.mineacademy.corearena.short. .dW.a(SourceFile:95)
[15:26:22 INFO]: at org.mineacademy.corearena.short. .dW.b(SourceFile:62)
[15:26:22 INFO]: at org.mineacademy.corearena.short. .dW.a(SourceFile:119)
[15:26:22 INFO]: at org.mineacademy.corearena.short. .fS.e(SourceFile:33)
[15:26:22 INFO]: at org.mineacademy.corearena.short. .eW$a.a(SourceFile:132)
[15:26:22 INFO]: at org.mineacademy.corearena.short. .aD.a(SourceFile:775)
[15:26:22 INFO]: at org.mineacademy.corearena.short. .aE.onMenuClick(SourceFile:75)
[15:26:22 INFO]: at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor1115.execute(Unknown Source)
[15:26:22 INFO]: at org.bukkit.plugin.EventExecutor.lambda$create$1(EventExecutor.java:69)
[15:26:22 INFO]: at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:80)
[15:26:22 INFO]: at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70)
[15:26:22 INFO]: at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:607)
[15:26:22 INFO]: at java.lang.Thread.run(Unknown Source)
[15:26:22 INFO]: !-----------------------------------------------------!
[15:26:22 INFO]: Error clicking in menu eW{}
[15:26:22 INFO]: !-----------------------------------------------------!
```

Also another issue while testing, I was not able to use the "procedural damage bow" because the game says that I'm not allowed to run the /ma items during a game even though I have added it to the allowed commands and tested it with other commands too. For some reason its impossible to run /ma items during the game and I can not understand why? Firstly I would like to know how to get procedural damage gravity engine working because I'm completly stuck. I have been able to get the initial and final snapshots working fine with AsyncWorldEdit as mentioned but I just cant get the gravity and the "Physical engine" working.

I am using version "FastAsyncVoxelSniper-flattened-1.0.3-backward" and "FastAsyncWorldEdit-1.16-311" which I got from 
https://intellectualsites.github.io/download/favs.html

Thanks so much for your help,
Waiting for your reply.

**Comment by LeehamElectronics** [comment]
I also tried this with only "FastAsyncWorldEdit-1.16-311" installed instead of using voxel sniper as well (whatever that means) and I got the exact same issue

**Comment by kangarko** [comment]
Please upload me your settings.yml.

Unfortunately it looks like FAWE is unsupported. Have you tried both AsyncWorldEdit and FastAsync ones? These are two separate distros.

**Comment by LeehamElectronics** [comment]
> Please upload me your settings.yml.
> 
> 
> 
> Unfortunately it looks like FAWE is unsupported. Have you tried both AsyncWorldEdit and FastAsync ones? These are two separate distros.

Thanks for the quick reply, yes I have tried both plugins and the map damage works fine with AsyncWE but the "gravity" and flying blocks doesn't work. I'll send the settings to u ASAP. Also any ideas on why /ma items doesn't work?
Cheers

**Comment by LeehamElectronics** [comment]
Here is the settings file:

[settings.zip](https://github.com/kangarko/CoreArena/files/5078294/settings.zip)

**Comment by LeehamElectronics** [comment]
Perhaps its my version of WorldEdit? I am using WorldEdit 7.2.0 beta 03 when using AsyncWorldEdit 3.7.5, would this be why gravity is not working on CoreArena? What version of World Edit would you recommend I use with AsyncWorldEdit?

**Comment by kangarko** [comment]
Maybe, can you please try experimenting with different WE / AWE versions? Yeah the gravity/flying blocks can be broken due to how it is handled async in these plugins in combination with CoreArena, unfortunately it doesn't appear to be easily fixable at the moment (unless I write my own AWE haha)

**Comment by LeehamElectronics** [comment]
I'm trying a combination of different versions of world edit and async world edit right now and I'll let u know how it goes. What version are you using to get it working? Also any ideas on why I can't run /ma items "in-game"? Haha yeh I wouldn't put that much pressure on you to re write all of FAWE lol

**Comment by kangarko** [comment]
/ma items works for me:
![ezgif-4-c414419d7fdf](https://user-images.githubusercontent.com/1918426/90308902-c444b100-dee3-11ea-8382-30bb10c82e08.gif)

Please test using the next version.

Please try getting AWE combination working on your machine, that is 1.15.2.

**Comment by LeehamElectronics** [comment]
Okay so I got nearly everything working with worldedit 7.2.0 beta 01 and AsyncWorldEditBossBar and AsyncWorldEdit-3.7.3
However the "gravity" and flying blocks engine only works on sand and naturally falling blocks, any idea why? Also the /ma items command works before you enter the arena, but as soon as you run /ma join the items command does not work. I can confirm that I have permissions and I have also enabled the command in the settings.yml
So basically my question is how can I enable the physics engine on blocks other than just sand? and how can I fix the "ma items" command?
![image](https://user-images.githubusercontent.com/51737378/90310195-a5b9d680-df32-11ea-843b-6a8c8c214e3a.png)
![image](https://user-images.githubusercontent.com/51737378/90310197-c5e99580-df32-11ea-9f37-c091cd522476.png)

Cheers

**Comment by kangarko** [comment]
Regarding /ma items, you are right, I had a debug mode on. Anyways, I added a new permission (see upcoming changelog) that will enable you to run any commands.

Regarding the physical engine, honestly I don't know, must be a problem in AsyncWorldEdit, so sorry about that!

**Comment by kangarko** [comment]
Are other blocks flying without Async worldedit, just on the normal version?

**Comment by LeehamElectronics** [comment]
What do you mean? Should I try the physical engine without ASyncWorledEdit and only use WorldEdit? I will give it a go now. What do you mean by " I added a new permission (see upcoming changelog)" wheres the changelog? Does that mean it will be fixed in the next update?
Cheers & thx for your quick replies I appreciate it.

**Comment by LeehamElectronics** [comment]
I tried using the physical engine again with only WorldEdit installed and still the same issue, although all of the block types do "break"... only sand / gravel actually has gravity / flies in the air. I also tried using AWE with the following config changes and that also did not work:
![gravity_AWE](https://user-images.githubusercontent.com/51737378/90311128-54faab80-df3b-11ea-8b71-e409fa5f52c5.PNG)
Is there a version of FAWE that you can confirm works with physical engine? I'm just desperate to get it working lol. If not could we have a rough ETA on when the physical engines "gravity / block flying" will be working with all block typed? 
No pressure.
Thanks

**Comment by kangarko** [comment]
I will publish a new version of CoreArena soon, there you will see the change log :)

Regarding the physics, I got it working: https://www.loom.com/share/43b247fc94844a8987cfd70016b9abd4 with just these plugins: https://i.imgur.com/432T8Gz.png

**Comment by LeehamElectronics** [comment]
Yay! I got procedural damage working with gravity using WorldEdit 7.2.0 beta 01 and AWE 3.7.3 and it works really well! Except... The gravity and flying blocks only occur with the Explosive Bow, is it possible to make creeper explosions and TNT have the same effect as the Explosion Bow? For example it would be nice if a creeper blows up near a building and then the building falls down due to the procedural damage gravity. Also I get massive lag spikes when joining an arena like the DAF map. Any ideas? Thanks

**Comment by kangarko** [comment]
Thanks, that can be a nice suggestion, please open a new ticket for that :)

**Comment by LeehamElectronics** [comment]
> Thanks, that can be a nice suggestion, please open a new ticket for that :)


https://github.com/kangarko/CoreArena/issues/327

Thanks!
